rpc/call_raw.go

```diff
@@ -111,11 +113,28 @@ func (c *Client) callBatchMethods(ctx context.Context, msgs json.RawMessage) str
 // callSingleMethod executes single JSON-RPC message and constructs proper response.
 func (c *Client) callSingleMethod(ctx context.Context, msg json.RawMessage) string {
 	// unmarshal JSON body into json-rpc request
-	chainID, method, params, id, err := methodAndParamsFromBody(msg)
+	jsonrpcParams, err := methodAndParamsFromBody(msg)
 	if err != nil {
-		return newErrorResponse(errInvalidMessageCode, err, id)
+		return newErrorResponse(errInvalidMessageCode, err, nil)
+	}
+
+	chainID := jsonrpcParams.ChainID
+	method := jsonrpcParams.Method
+	params := jsonrpcParams.Params
+	timeout := jsonrpcParams.Timeout
+	id := jsonrpcParams.ID
+
+	if timeout != 0 {
```

Comment:
We could check if timeout is positive to be a tad more strict.

Reply:
it's an uint, so in theory it can't be negative, or you mean in case we change it to be an int?

Reply:
Perfect then, I just skipped over the type definition and thought it was an int.